axelrod/tests/strategies/test_player.py

```diff
@@ -245,27 +246,56 @@ def test_reset_clone(self):
         clone = player.clone()
         self.attribute_equality_test(player, clone)
 
+    def test_equal(self):
+        """Test the player's `__eq__` function."""
+        player1 = Cooperator()
+        player2 = Cooperator()
+        self.assertEqual(player1, player1)
+        #Test with a difference in name attribute
```

Comment:
These look good but could you break them up in to individual test methods (more or less according to your inline comments):

```
def test_equal(self):
    player1 = Cooperator()
    player2 = Cooperator()
    self.assertEqual(player1, player1)

def test_not_equal_name(self):
...

def test_not_equal_classifier(self):
...
```

This helps with understanding potential failures.